src/main/resources/templates/task.html

```diff
@@ -1,59 +1,57 @@
+            <div class="tab-content" th:inline="text">        
+                <div th:each="editor, iter: ${editorContents}" th:id="${editor.key}+'_item'" th:class="${iter.index == 0}?'panel panel-default tab-pane fade in active':'panel panel-default tab-pane fade'">
+                    <div class="panel-heading">
+                        <h3 class="panel-title" th:inline="text">[[${editor.value[0]}]]</h3>
+                    </div>
+                    <div class="panel-body">
+                        <form method="post" action="/task">
+                            <input type="hidden" name="challenge" th:value="${challengeId}" />
+                            <input type="hidden" name="task" th:value="${taskId}" />
+                            <div th:id="${editor.key}" style="height: 300px" th:inline="text">[[${editor.value[1]}]]</div>
+                            <textarea name="code" th:id="'shadow-'+${editor.key}" style="display: none;"/>
+                            <button type="submit" class="btn btn-primary">Submit</button>
+                        </form>
+                    </div>
+                </div>
+            </div>
         </div>
-        <div class="col-md-6">
-            <p th:text="${taskdesc}"></p><br/><br/>
-        </div>
-    </div>
-
-    <div th:if="${errors}" class="row">
-        <p>The following errors were encountered:</p>
-        <ul>
-            <li th:each="error : ${errors}">
-                <span th:text="${error}"></span>
-            </li>
-        </ul>
-    </div>
-
-    <div class="panel panel-default">
-        <div class="panel-heading">
-            <h3 class="panel-title">Editor</h3>
-        </div>
-        <div class="panel-body">
-            <form method="post" action="/task">
-                <input type="hidden" name="challenge" th:value="${challengeId}" />
-                <input type="hidden" name="task" th:value="${taskId}" />
-                <div id="editor" style="height: 300px" th:text="${code}"></div>
-                <textarea name="code" id="shadow-editor" style="display: none;"/>
-                <button type="submit" class="btn btn-primary">Submit</button>
-            </form>
-        </div>
-    </div>
-</div>
-
-<script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.2.8/ace.js" type="text/javascript" charset="utf-8"></script>
-<script>
-    var editor = ace.edit("editor");
-    editor.setTheme("ace/theme/monokai");
-    editor.getSession().setMode("ace/mode/java");
-
-    var textarea = document.getElementById("shadow-editor");
-    editor.getSession().on("change", function () {
-        textarea.value = editor.getSession().getValue();
-    });
-</script>
-
-</body>
-</html>
\ No newline at end of file
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.2.8/ace.js" type="text/javascript" charset="utf-8"></script>
+        <script th:inline="text" th:each="editor: ${editorContents}">
+            var [[${editor.key}]] = ace.edit("[[${editor.key}]]");
```

Comment:
No need to generate JS here. Giving the editors a separate CSS class and looping through the result of `document.getElementsByClassName()` can be used here.